edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesServiceBuilder.cs

```diff
@@ -0,0 +1,163 @@
+                }
+
+                // Handle HostConfig PortBindings entries
+                if (moduleWithDockerConfig.Config?.CreateOptions?.HostConfig?.PortBindings != null)
+                {
+                    foreach (KeyValuePair<string, IList<DockerModels.PortBinding>> portBinding in moduleWithDockerConfig.Config?.CreateOptions?.HostConfig?.PortBindings)
+                    {
+                        string[] portProtocol = portBinding.Key.Split('/');
+                        if (portProtocol.Length == 2)
+                        {
+                            if (int.TryParse(portProtocol[0], out int port) && ProtocolExtensions.TryValidateProtocol(portProtocol[1], out string protocol))
+                            {
+                                // Entries in Docker portMap wants to expose a port on the host (hostPort) and map it to the container's port (port)
+                                // We interpret that as the pod wants the cluster to expose a port on a public IP (hostPort), and target it to the container's port (port)
+                                foreach (DockerModels.PortBinding hostBinding in portBinding.Value)
+                                {
+                                    if (int.TryParse(hostBinding.HostPort, out int hostPort))
+                                    {
+                                        // If a port entry contains the same "port", then remove it and replace with a new ServicePort that contains a target.
+                                        var duplicate = portList.SingleOrDefault(a => a.Port == hostPort);
+                                        if (duplicate != default(V1ServicePort))
+                                        {
+                                            portList.Remove(duplicate);
+                                        }
+
+                                        portList.Add(new V1ServicePort(hostPort, name: $"HostPort-{port}-{protocol.ToLower()}", protocol: protocol, targetPort: port));
+                                        onlyExposedPorts = false;
+                                    }
+                                    else
+                                    {
+                                        Events.PortBindingValue(module, portBinding.Key);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (portList.Count > 0)
+            {
+                // Selector: by module name and device name, also how we will label this puppy.
+                var objectMeta = new V1ObjectMeta(annotations: serviceAnnotations.GetOrElse(() => null), labels: labels, name: KubeUtils.SanitizeDNSValue(moduleIdentity.ModuleId));
```

Comment:
You could you `Option<T>.OrDefault()` method instead of `GetOrElse(() => default(T))`